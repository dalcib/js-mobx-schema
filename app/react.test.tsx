import * as React from 'react'
import Model from './model'
import ClassSchemaExample from './example'
import * as ReactTestRenderer from 'react-test-renderer'
import { observer } from 'mobx-react'

//jest.mock('mobx-react', () => require('mobx-react/custom'))

type ObjectFields<T> = { [P in keyof T]: T[P] }

const f = new Model(ClassSchemaExample)
const l = f.l

const Comp = observer(() =>
  <div>
    <input {...l('name')} />
    {JSON.stringify(f.errorsMessages)}
    {f.errorsText}
  </div>
)

//{JSON.stringify(f.errors)}

const CompOb = (
  <div>
    <Comp />
  </div>
)

const renderer = ReactTestRenderer.create(CompOb)

describe('component', () => {
  it('test renderer', () => {
    expect(renderer).toMatchSnapshot()
    f.data.name = 'aaaaa'
    expect(renderer).toMatchSnapshot()
    f.data.firstName = 'aa'
    expect(renderer).toMatchSnapshot()
    f.data.firstName = 'aacccccc'
    expect(renderer).toMatchSnapshot()
  })
})
