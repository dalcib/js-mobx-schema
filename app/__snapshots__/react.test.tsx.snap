// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component test renderer 1`] = `
<div>
  <div>
    <input
      onChange={[Function]}
      value="fghkkk"
    />
    {"firstName":"is a required property","lastName":"is a required property"}
    data.firstName is a required property, data.lastName is a required property
  </div>
</div>
`;

exports[`component test renderer 2`] = `
<div>
  <div>
    <input
      onChange={[Function]}
      value="aaaaa"
    />
    {"firstName":"is a required property","lastName":"is a required property"}
    data.firstName is a required property, data.lastName is a required property
  </div>
</div>
`;

exports[`component test renderer 3`] = `
<div>
  <div>
    <input
      onChange={[Function]}
      value="aaaaa"
    />
    {"firstName":"should NOT be shorter than 5 characters","lastName":"is a required property"}
    data.firstName should NOT be shorter than 5 characters, data.lastName is a required property
  </div>
</div>
`;

exports[`component test renderer 4`] = `
<div>
  <div>
    <input
      onChange={[Function]}
      value="aaaaa"
    />
    {"lastName":"is a required property"}
    data.lastName is a required property
  </div>
</div>
`;
